layers/+distributions/spacemacs-base/local/centered-buffer-mode/centered-buffer-mode.el

```diff
@@ -46,82 +46,136 @@ that differed modifications won't cause an overflow."
-                        left-fringe-width fringe-w
-                        right-fringe-width fringe-w
-                        ;; looks better with some margin.
-                        left-margin-width (if (or (not left-margin-width)
-                                                  (= left-margin-width 0))
-                                              1
-                                            left-margin-width)
-                        spacemacs--centered-buffer-mode-text-pixel-size
-                        (car (window-text-pixel-size window)))
-                  (face-remap-add-relative 'fringe :background
-                                           spacemacs-centered-buffer-mode-fringe-color)
-                  (set-window-buffer window indirect-buffer)
-                  (add-hook 'buffer-list-update-hook
-                            'spacemacs//centered-buffer-buffer-list-update-fringes)
-                  (add-hook 'window-configuration-change-hook
-                            'spacemacs//centered-buffer-buffer-list-update-fringes))
-              (setq spacemacs--centered-buffer-mode-origin-buffer nil)
-              (set-buffer origin-buffer)
-              (kill-buffer indirect-buffer)
-              (setq spacemacs--centered-buffer-mode-indirect-buffer nil)
-              (when (called-interactively-p 'any)
-                (message "Not enough space to center the buffer!"))))))
-    (let* ((window (selected-window))
-           (origin-buffer spacemacs--centered-buffer-mode-origin-buffer)
-           (indirect-buffer (window-buffer window)))
-      (setq spacemacs--centered-buffer-mode-origin-buffer nil)
-      (when (buffer-live-p origin-buffer)
-        (switch-to-buffer origin-buffer nil t)
-        (setq spacemacs--centered-buffer-mode-indirect-buffer nil))
-      (when (buffer-live-p indirect-buffer)
-        (dolist (window (get-buffer-window-list indirect-buffer 2))
-          (set-window-buffer window origin-buffer))
-        (kill-buffer indirect-buffer)))))
+            (setq spacemacs--centered-buffer-mode-indirect-buffer
+                  (make-indirect-buffer origin-buffer
+                                        (format "%s(centered)"
+                                                origin-buffer)
+                                        t)))))
+    (spacemacs//centered-buffer-mode-hide-buffer-toggle origin-buffer t)
+    (spacemacs//centered-buffer-mode-buffer-fringr-color-toggle origin-buffer t)
+    ;; Mode will be applied to the indirect buffer.
+    (setq spacemacs-centered-buffer-mode
+          nil
+          spacemacs--centered-buffer-mode-indirect-buffer
+          indirect-buffer)
+    (when (derived-mode-p 'org-mode)
+      (setq-local org-startup-folded nil)
+      (outline-show-all))
+    (switch-to-buffer indirect-buffer nil t)
+    (with-mode-disabled page-break-lines-mode
+                        (let* ((fringe-w (spacemacs//centered-buffer-calc-fringe
```

Comment:
better to put `page-break-lines-mode` on a new line to avoid the huge indentation.